controllers/repo_manager/controller.go

```diff
@@ -284,7 +294,7 @@ func (r *RepoManagerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			} else if pulpController.Requeue {
 				return pulpController, nil
 			}
-		} else if strings.ToLower(pulp.Spec.IngressType) == "ingress" {
+		} else if r.isNginxIngress(pulp) {
```

Comment:
Sorry if my understanding is incorrect, but when non-Nginx ingress controller is being used, is ingress resources no longer deployed by this change?

Reply:
> "[..] when non-Nginx ingress controller is being used, is ingress resources no longer deployed by this change?"

Yes, if the ingress class defined in CR does not have Nginx as controller, no ingress resource will be deployed (the operator will keep deploying pulp-web resources instead).
But this is not a new behavior, since the release of ingress resource configuration the operator worked like that.

This PR will allow users to define the ingress class to avoid issues in environments with multiple or no default ingressclass(es).
We are also updating the documentation to give more information about why it works only with nginx controller: https://github.com/pulp/pulp-operator/blob/9bb31d1548f99f58f3968b8444bbf05f128e5eb1/docs/configuring/reverse_proxy.md

Reply:
> "actually, we need to deploy the ingress redirecting to pulp-web"

What if the ingress provided does not support rewrite?
What if the user wants tls termination on ingress? What if the user wants reencryption?

I added more details in #770.

Reply:
> What if the ingress provided does not support rewrite? 

the rewrite logic is inside pulp-web pod, so the ingress itself will only point to the web service as we did here:
https://github.com/pulp/pulp-operator/blob/ansible/roles/pulp-routes/templates/pulp.ingress.yaml.j2#L1-L30

> What if the user wants tls termination on ingress?

we did it here https://github.com/pulp/pulp-operator/blob/ansible/roles/pulp-routes/templates/pulp.ingress.yaml.j2#L24-L29

Reply:
> "the rewrite logic is inside pulp-web pod, so the ingress itself will only point to the web service as we did here"

Why do we need ingress in this case? Not sure if I am missing something.


> we did it here https://github.com/pulp/pulp-operator/blob/ansible/roles/pulp-routes/templates/pulp.ingress.yaml.j2#L24-L29"

this is the ansible branch.
Correct me if I'm wrong, but the current version of golang operator does not provide this configuration.
In case it is possible to configure tls termination, where/how do I configure my custom certificate? We should also close issue #676 since the feature is available already.

Reply:
we need the ingress to expose the web service,
and we do it here:
https://github.com/pulp/pulp-operator/blob/main/controllers/repo_manager/ingress.go#L239-L251